datafusion/physical-expr/src/partitioning.rs

```diff
@@ -15,14 +15,43 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! [`Partitioning`] and [`Distribution`] for physical expressions
+//! [`Partitioning`] and [`Distribution`] for `ExecutionPlans`
 
 use std::fmt;
 use std::sync::Arc;
 
 use crate::{expr_list_eq_strict_order, EquivalenceProperties, PhysicalExpr};
 
-/// Partitioning schemes supported by operators.
+/// Partitioning schemes supported by [`ExecutionPlan`]s.
+///
+/// A partition represents an independent stream that an `ExecutionPlan` can
+/// produce in parallel. Each `ExecutionPlan` must produce at least one
```

Comment:
Each `ExecutionPlan` instance must produce at least one partition, and consume at most one partition.

Reply:
An ExecutionPlan can consume one or more partitions. I'll try and clarify that later today

Reply:
yes please my understanding by trait `.execute()` signature is to run the method for 1 partition at most
```
    fn execute(
        &self,
        partition: usize,
        context: Arc<TaskContext>,
    ) -> Result<SendableRecordBatchStream>;
```